modules/gapi/misc/python/shadow_gapi.hpp

```diff
@@ -3,11 +3,10 @@
 
 namespace cv
 {
-   struct GAPI_EXPORTS_W_SIMPLE GCompileArg { };
-
-   GAPI_EXPORTS_W GCompileArgs compile_args(gapi::GKernelPackage pkg);
-   GAPI_EXPORTS_W GCompileArgs compile_args(gapi::GNetPackage pkg);
-   GAPI_EXPORTS_W GCompileArgs compile_args(gapi::GKernelPackage kernels, gapi::GNetPackage nets);
+   struct GAPI_EXPORTS_W_SIMPLE GCompileArg {
+       GAPI_WRAP GCompileArg(gapi::GKernelPackage pkg);
```

Comment:
To extend number of possible types in other module need to do:
```
// shadow_somemodule.hpp
struct GCompileArg {
    GAPI_WRAP GCompileArg(module::SomeArg arg);
};
```

Reply:
This code add new overload to `GCompileArg` ctor

Reply:
But how did you solve  variadic arguments passing?

Reply:
it's solved indirectly through python's `compile_args` (which probably requires a better PR name since it's confusing - vaargs is a thing in C++ as well).

`map()` calls a function for each element of a range. given that the function is a `GCompileArg` ctor, we convert "raw" args into a range containing `cv2.GCompileArg(arg)` elements (by the way, does ctor qualify as a function?)

Reply:
>>> by the way, does ctor qualify as a function

Why not ? https://stackoverflow.com/questions/11943831/python-how-to-put-constructors-in-map-function